lib/matplotlib/tests/test_backend_bases.py

```diff
@@ -349,3 +350,56 @@ def test_interactive_pan(key, mouseend, expectedxlim, expectedylim):
     # Should be close, but won't be exact due to screen integer resolution
     assert tuple(ax.get_xlim()) == pytest.approx(expectedxlim, abs=0.02)
     assert tuple(ax.get_ylim()) == pytest.approx(expectedylim, abs=0.02)
+
+
+def test_toolmanager_remove():
+    expected_warning_regex = (
+        r"Treat the new Tool classes introduced in "
+        r"v[0-9]*.[0-9]* as experimental for now; "
+        "the API and rcParam may change in future versions.")
```

Comment:
Maybe we should make this a file-level global `toolmanager_expected_warning(_regex)`?

Reply:
Not sure I know how to do this. Couldn't find any other example of this in the code-base. But at least moved the regex to a single global string, so a slight improvement.

Reply:
Not sure what you thought I meant, but that is what I meant.

Reply:
Apparently, this is a way to avoid it:
https://github.com/matplotlib/matplotlib/blob/36685ebfed67a594b910382eb1ba1db04c2338a8/lib/matplotlib/tests/test_widgets.py#L967-L968